src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -0,0 +1,272 @@
+/**
+ * @file StepAdaptationController.hpp
+ * @authors Milad Shafiee <milad.shafiee@iit.it>
+ * @copyright 2020 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2020
+ */
+
+#ifndef WALKING_CONTROLLERS_STEP_ADAPTATION_CONTROLLERS_H
+#define WALKING_CONTROLLERS_STEP_ADAPTATION_CONTROLLERS_H
+
+
+// std
+#include <memory>
+
+// iDynTree
+#include <iDynTree/Core/Triplets.h>
+#include <iDynTree/Core/SparseMatrix.h>
+#include <iDynTree/ConvexHullHelpers.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/Twist.h>
+#include <iDynTree/Core/SpatialAcc.h>
+
+// osqp-eigen
+#include <OsqpEigen/OsqpEigen.h>
+
+// yarp
+#include <yarp/os/Value.h>
+
+#include <unordered_map>
+
+#include <qpOASES.hpp>
+#include <WalkingControllers/iDynTreeUtilities/Helper.h>
+
+typedef struct{
+    double maxFootHeight;
+    double discretizationTime ;
+    double takeOffTime;
+    double yawAngleAtImpact;
+    iDynTree::Vector2 zmpToCenterOfFootPosition;
+    iDynTree::Transform currentFootTransform;
+    iDynTree::Twist currentFootTwist;
+    iDynTree::Transform adaptatedFootTransform;
+    iDynTree::Twist adaptedFootTwist;
+    iDynTree::SpatialAcc adaptedFootAcceleration;
```

Comment:
This is supposed to be the output right? I would remove them from here since they would be duplicated of the other parameters of ``getAdaptatedFootTrajectory``.

Reply:
Done!